kmath-core/src/commonMain/kotlin/scientifik/kmath/operations/NumberAlgebra.kt

```diff
@@ -54,93 +63,109 @@ inline class Real(val value: Double) : FieldElement<Double, Real, RealField> {
 
-    override inline fun exp(arg: Float) = kotlin.math.exp(arg)
-    override inline fun ln(arg: Float) = kotlin.math.ln(arg)
+    override inline fun multiply(a: Float, b: Float): Float = a * b
 
-    override inline fun norm(arg: Float) = abs(arg)
+    override inline fun divide(a: Float, b: Float): Float = a / b
 
-    override inline fun Float.unaryMinus() = -this
+    override inline fun sin(arg: Float): Float = kotlin.math.sin(arg)
+    override inline fun cos(arg: Float): Float = kotlin.math.cos(arg)
+    override inline fun tan(arg: Float): Float = kotlin.math.tan(arg)
+    override inline fun acos(arg: Float): Float = kotlin.math.acos(arg)
+    override inline fun asin(arg: Float): Float = kotlin.math.asin(arg)
+    override inline fun atan(arg: Float): Float = kotlin.math.atan(arg)
 
-    override inline fun Float.plus(b: Float) = this + b
+    override inline fun sinh(arg: Float): Float = kotlin.math.sinh(arg)
+    override inline fun cosh(arg: Float): Float = kotlin.math.cosh(arg)
+    override inline fun tanh(arg: Float): Float = kotlin.math.tanh(arg)
+    override inline fun asinh(arg: Float): Float = kotlin.math.asinh(arg)
+    override inline fun acosh(arg: Float): Float = kotlin.math.acosh(arg)
+    override inline fun atanh(arg: Float): Float = kotlin.math.atanh(arg)
 
-    override inline fun Float.minus(b: Float) = this - b
+    override inline fun power(arg: Float, pow: Number): Float = arg.kpow(pow.toFloat())
+    override inline fun exp(arg: Float): Float = kotlin.math.exp(arg)
+    override inline fun ln(arg: Float): Float = kotlin.math.ln(arg)
 
-    override inline fun Float.times(b: Float) = this * b
+    override inline fun norm(arg: Float): Float = abs(arg)
 
-    override inline fun Float.div(b: Float) = this / b
+    override inline fun Float.unaryMinus(): Float = -this
+    override inline fun Float.plus(b: Float): Float = this + b
+    override inline fun Float.minus(b: Float): Float = this - b
+    override inline fun Float.times(b: Float): Float = this * b
+    override inline fun Float.div(b: Float): Float = this / b
 }
 
 /**
  * A field for [Int] without boxing. Does not produce corresponding field element
  */
 @Suppress("EXTENSION_SHADOWED_BY_MEMBER", "OVERRIDE_BY_INLINE", "NOTHING_TO_INLINE")
 object IntRing : Ring<Int>, Norm<Int, Int> {
-    override val zero: Int = 0
-    override inline fun add(a: Int, b: Int) = a + b
-    override inline fun multiply(a: Int, b: Int) = a * b
-    override inline fun multiply(a: Int, k: Number) = k.toInt() * a
-    override val one: Int = 1
+    override val zero: Int
+        get() = 0
 
-    override inline fun norm(arg: Int) = abs(arg)
+    override val one: Int
+        get() = 1
 
-    override inline fun Int.unaryMinus() = -this
+    override inline fun add(a: Int, b: Int): Int = a + b
+    override inline fun multiply(a: Int, k: Number): Int = k.toInt() * a
 
-    override inline fun Int.plus(b: Int): Int = this + b
```

Comment:
I do not think that removing this one is free. It could hinder inlining. The same goes for all similar cases. It should not be done without benchmarks.

Reply:
I'll revert it